### PR TITLE
feat: impl cluster buffer limit

### LINKED ISSUE #3278: Add a knob to make per_connection_buffer_limit_bytes of cluster configurable
*Description*:
> EG can configure `per_connection_buffer_limit_bytes` on xDS Listener currently via #2600, not available to configure `per_connection_buffer_limit_bytes` on xDS Cluster, default to `32KB`. A new knob will be better for entire buffer sizing control of  Envoy provisioned by EG.

[optional *Relevant Links*:]
> https://gateway.envoyproxy.io/latest/api/extension_types/#connection
> https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-cluster

### PR DESCRIPTION
**Pull request by ShyunnY**
**What type of PR is this?**

_type_: 
feat impl cluster buffer limit

**What this PR does / why we need it**:

_description_: 
Support `per_connection_buffer_limit_bytes` configuration on cluster by adding a knob

> ref doc: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-cluster

**Which issue(s) this PR fixes**:

Fixes #3278

### CODE REVIEW

File: internal/gatewayapi/testdata/backendtrafficpolicy-buffer-limit-with-error.out.yaml
Comment (shawnh2) [inline internal/gatewayapi/testdata/backendtrafficpolicy-buffer-limit-with-error.out.yaml:1]: about the naming here, `-with-invalid-value` sounds better than just `-with-error`

**Review by shawnh2 — COMMENTED** [review]
LGTM! Just one nit.

File: internal/ir/xds.go [hunk 1]
```diff
@@ -2024,6 +2030,13 @@ type TLSUpstreamConfig struct {
 	TLSConfig           `json:",inline"`
 }
 
+// BackendPolicyConnection settings for upstream connections
+// +k8s:deepcopy-gen=true
+type BackendPolicyConnection struct {
```
Comment (arkodg) [inline internal/ir/xds.go:2035]: prefer `backendConnection`

Comment (ShyunnY) [reply internal/ir/xds.go]: Done!

**Review by AliceProxy — APPROVED** [review]

**Pull request merged**